plugin/lighthouse/handler.go

```diff
@@ -91,12 +152,18 @@ func (lh *Lighthouse) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
 
 	records := make([]dns.RR, 0)
 
-	for _, ip := range ips {
-		record := &dns.A{Hdr: dns.RR_Header{Name: state.QName(), Rrtype: dns.TypeA, Class: state.QClass(), Ttl: lh.ttl}, A: net.ParseIP(ip).To4()}
-		log.Debugf("rr is %v", record)
-		records = append(records, record)
+	if state.QType() == dns.TypeA {
+		records = lh.createARecords(ips, state)
+	} else if state.QType() == dns.TypeSRV {
+		records = lh.createSRVRecords(record, state, pReq, zone)
+		if records == nil {
```

Comment:
Perhaps this should be moved after L165?

```suggestion
		if len(records) {
```